runtime/libia2/init.c

```diff
@@ -32,8 +32,12 @@ char *allocate_stack(int i) {
   /* Tag the allocated stack pointer so it is accessed with the right pkey */
   stack = (char *)((uint64_t)stack | (uint64_t)i << 56);
 #endif
+#ifdef __aarch64__
+  return stack + STACK_SIZE - 16;
```

Comment:
Is there a good reason to not also do this on x86?

Reply:
This merged as part of #496: https://github.com/immunant/IA2-Phase2/pull/496/commits/5a6fc059c6ea1ed21775280d91c1cb2fc30a05be